src/core/CoolBoard.cpp

```diff
@@ -377,11 +378,71 @@ void CoolBoard::readBoardData(JsonObject &reported) {
   }
 }
 
-void CoolBoard::sleep(unsigned long interval) {
-  if (interval > 0) {
-    INFO_VAR("Going to sleep for seconds:", interval);
-    ESP.deepSleep((uint64(interval) * 1000000ULL), WAKE_RF_DEFAULT);
+void CoolBoard::shiftLoginterval(uint32_t value) {
+  uint8_t val[4];
+  val[0] = value - MAX_SLEEP_TIME;
+  val[1] = ((value - MAX_SLEEP_TIME) >> 8);
+  val[2] = ((value - MAX_SLEEP_TIME) >> 16);
+  val[3] = ((value - MAX_SLEEP_TIME) >> 24);
+  EEPROM.write(0x00, val[0]);
+  EEPROM.write(0x01, val[1]);
+  EEPROM.write(0x02, val[2]);
+  EEPROM.write(0x03, val[3]);
+  EEPROM.end();
+  SPIFFS.end();
+}
+
+void CoolBoard::sleep() {
+  EEPROM.begin(5);
+  uint32_t value = 0;
+  delay(5000);
+  if (!SPIFFS.exists("/configEEPROM.flag") ||
+      digitalRead(BOOTSTRAP_PIN) == LOW) {
+    for (int i = 0; i < 5; i++) {
+      EEPROM.write(i, 0);
+    }
+    File f;
+    if (!(f = SPIFFS.open("/configEEPROM.flag", "w"))) {
+      ERROR_LOG("Can't create file configEEPROM.flag in SPIFFS");
+    } else {
+      f.close();
+    }
+  }
+  if ((value = (EEPROM.read(0x00)))) {
```

Comment:
No, this is dangerous, you should test the whole value, parts of if might me zero (for instance, if we need to sleep 256 seconds)